src/viam/examples/dial/example_dial.cpp

```diff
@@ -68,7 +68,5 @@ int main() {
         std::cout << "got generic client named " << gc->name() << std::endl;
     }
 
-    robot->close();
```

Comment:
Spoke with @stuqdog offline. Robot clients are closed in the types constructors, so this line leads to a double close and a "broken pipe" error; we should remove it.

Reply:
Does that mean that the `close` method on `RobotClient` is dangerous and should be removed? Or, should it be ok to `close` a `RobotClient`, and the dtor for `RobotClient` should deal with it more gracefully?

Reply:
It's a good question; @stuqdog would know more about the semantics of `close()`. Judging from [the doc comments](https://github.com/viamrobotics/viam-cpp-sdk/blob/main/src/viam/sdk/robot/client.hpp#L42-L44), the necessity of calling `close` seems to depend on which way you've constructed the `RobotClient`.

Reply:
That's correct. The thinking is/was, if people are asking us to build a channel for them (via the `at_address` call) then it makes sense for us to `close` that channel when the client is done with it. If a user is providing their own channel (via the `with_channel` call) then we don't want to make assumptions about that when to close that channel.

That said, I do think we may be painting with too broad a brush here. Just because the channel isn't being closed doesn't necessarily mean we shouldn't call the other aspects of `close` (namely, the `~thread()` calls). So I do think there's room for making `close` cleaner/making the `RobotClient` dtor to be more graceful in its handling of it.

Reply:
Gotcha; thanks for the context! Filed https://viam.atlassian.net/browse/RSDK-6212, as it seems slightly out of scope for this PR.